app/api/server_test.go

```diff
@@ -0,0 +1,79 @@
+package api
+
+import (
+	"encoding/xml"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/umputun/feed-master/app/config"
+	"github.com/umputun/feed-master/app/feed"
+	"github.com/umputun/feed-master/app/proc"
+	bolt "go.etcd.io/bbolt"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestGetFeedCtrl(t *testing.T) {
+	tmpfile := filepath.Join(os.TempDir(), "test.db")
+	defer os.Remove(tmpfile)
+	testDB, err := bolt.Open(tmpfile, 0o600, nil)
+	require.NoError(t, err)
+	boltStore := &proc.BoltDB{DB: testDB}
+
+	conf := &config.Conf{}
+	conf.System.BaseURL = "http://localhost:8080"
+	conf.System.MaxTotal = 3
+	conf.YouTube.BaseURL = ""
+
+	srv := Server{
+		Store: boltStore,
+		Conf:  *conf,
+	}
+	ts := httptest.NewServer(srv.router())
+	defer ts.Close()
+
+	item := feed.Item{
+		Title:       "test",
+		Link:        "http://test.com",
+		Description: "Description",
+		GUID:        "GUID123",
+		Duration:    "123",
+		PubDate:     "Mon, 02 Jan 2006 15:04:05 -0700",
+	}
+	created, err := srv.Store.Save("test", item)
+	require.NoError(t, err)
+	assert.True(t, created)
+
+	rr, err := ts.Client().Get(ts.URL + "/rss/test")
+	require.NoError(t, err)
+	defer rr.Body.Close()
+
+	body := make([]byte, rr.ContentLength)
```

Comment:
i don't see any direct use of body here, except unmarshal. Reading the response makes sense if some others/additional checks on the body needed, for example converting body to string and assert for contains some expected elements. However, if the only use is unmarshal it can be done directly with response body reader